bootcompiler/src/main/scala/org/mozartoz/bootcompiler/transform/Desugar.scala

```diff
@@ -187,6 +187,25 @@ object Desugar extends Transformer with TreeDSL {
       treeCopy.MatchStatement(expr, value,
           clauses map(c => matchExpressionClauseToBindStatement(result, c)),
           exprToBindStatement(result, elseExpression))
+
+    case Record(label, fields) =>
+      val lastI = fields.lastIndexWhere(_.value match {
+        case CallExpression(callable, args) => true
+        case _                              => false
+      })
```

Comment:
This is kind of arbitrary, since we can only optimize one of them. I could have chosen the first, or any, but I think developers are aware this is usually the last call that is optimized.
I still think it would be interesting to have a look at the book, see if they tell anything about that. I'm going to try to borrow one tomorrow.